### PR TITLE
Add StackBlitz example for macros

### PR DESCRIPTION
**Pull request by SamVerschueren**
This PR adds an `examples` directory with currently one example, being `macros`.

I've copied one macro case from the documentation as you will see. We can add a different one, or add multiple test files even. But maybe, the smaller the example, the easier it is to understand for new people. Open to suggestions here.

If you want to test it out how it will look like if you click the button, use this button which does the same but points to my fork.

[Open in StackBlitz](https://stackblitz.com/github/SamVerschueren/ava/tree/stackblitz/examples/macros/examples/macros?file=test.js&terminal=test&view=editor)

These are the properties I added to the URL
- Open `test.js` by default if you click the button
- Run the `test` command by default if someone opens the example
- Hide the preview pane on the right

One small sidenote, I did notice that the `view=editor` query parameter is not working and the preview pane is still open, but I will look into that to fix it asap.

Some small questions
- Do you guys want to use the `Open in StackBlitz` button? Or do you rather prefer textual links?
- Is it ok to embed the buttons in the `docs` documentation? I assume that the easier it is to play with a feature, the better. Then people don't have to skim through the examples.
- Do we want to put more information in the `readme.md` of the example? This is what it looks like now https://github.com/SamVerschueren/ava/tree/stackblitz/examples/macros/examples/macros. I didn't want to copy parts of the documentation because it can only become outdated.

### CODE REVIEW

**Comment by novemberborn** [comment]
I think this all great. The button is a bit attention grabby but that's fine for now.

We may want to come up with a better example though. XO rejects it.

**Comment by SamVerschueren** [comment]
Yeah noticed this as well. Maybe just a `sum` method in index.js?

**Comment by novemberborn** [comment]
Sounds good.

If you merge in `main` CI should pass again.

File: examples/macros/test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+const test = require('ava');
+const { sum } = require('.');
+
+function macro(t, a, b, expected) {
+	t.is(sum(a, b), expected);
+}
+
+macro.title = (providedTitle = '', a, b, expected) => `${providedTitle} ${a}+${b} = ${expected}`.trim();
```
Comment (SamVerschueren) [inline examples/macros/test.js:9]: @novemberborn linting fails on the `providedTitle = ''`, you want me to remove it and use `${providedTitle || ''}` instead?

Comment (SamVerschueren) [reply examples/macros/test.js]: I removed the default value and used `|| ''`.

**Comment by SamVerschueren** [comment]
When clicking the button, you will now see that the preview window is gone and the editor takes the entire screen. I fixed that issue upstream.

**Comment by novemberborn** [comment]
Nice, thanks @SamVerschueren!

**Pull request merged**